packages/types/src/codec/EnumType.ts

```diff
@@ -25,62 +25,83 @@ type Aliasses = {
 };
 
 /**
- * @name EnumType
+ * @name Enum
  * @description
  * This implements an enum, that based on the value wraps a different type. It is effectively
  * an extension to enum where the value type is determined by the actual index.
  */
 // TODO:
 //   - As per Enum, actually use TS enum
 //   - It should rather probably extend Enum instead of copying code
-export default class EnumType<T> extends Base<Codec> implements Codec {
+export default class Enum extends Base<Codec> implements Codec {
   private _def: TypesDef;
   private _index: number;
   private _indexes: Array<number>;
+  private _isBasic: boolean;
 
-  constructor (def: TypesDef, value?: any, index?: number | EnumType<T>, aliasses?: Aliasses) {
-    const decoded = EnumType.decodeEnumType(def, aliasses || {}, value, index);
+  constructor (def: TypesDef | Array<string>, value?: any, index?: number | Enum, aliasses?: Aliasses) {
```

Comment:
Can we remove aliasses? It's used in one place (Finalised -> Finalized) and imo it's a bit useless now

Reply:
Good point, meant to look at those since I couldn't remember the use.